### ISSUE TITLE
[Develop] vdisk cache quota is broken

### ISSUE THREAD

**Issue by JeffreyDevloo** [openvstorage/framework#2025]
vdiskcheckup: the way the quota info is fetched is faulty

2018-04-12 14:33:12 73800 +0200 - DEV-4N-199-191 - 2258/140569954739968 - lib/vdisk.py - vdisk_checkup - 29 - ERROR - Error when setting cache quotas
Traceback (most recent call last):
  File "/opt/OpenvStorage/ovs/lib/vdisk.py", line 150, in vdisk_checkup
    metadata = vpool.metadata['backend']['caching_info'].get(vdisk.storagerouter_guid, {})
KeyError: 'caching_info'

Quota value seems to be 'False' instead of 'None'
Develop:
```
{u'607d892b-203c-4101-8309-94c3741771e6': {u'block_cache': {u'backend_info': {u'alba_backend_guid': u'f6606dcf-7b8b-4842-b332-a85a33b5e4ed',
    u'arakoon_config': {u'21Lvu16mKG7Lt3We': {u'client_port': 26408,
      u'crash_log_sinks': u'console:',
      u'fsync': u'true',
      u'home': u'/mnt/hdd2/arakoon/mybackend02-abm/db',
      u'ip': u'10.100.199.191',
      u'log_level': u'info',
      u'log_sinks': u'console:',
      u'messaging_port': 26409,
      u'name': u'21Lvu16mKG7Lt3We',
      u'tlog_compression': u'snappy',
      u'tlog_dir': u'/mnt/hdd2/arakoon/mybackend02-abm/tlogs'},
     u'Pxps6d1C7kZ8ZuyN': {u'client_port': 26408,
      u'crash_log_sinks': u'console:',
      u'fsync': u'true',
      u'home': u'/mnt/hdd2/arakoon/mybackend02-abm/db',
      u'ip': u'10.100.199.192',
      u'log_level': u'info',
      u'log_sinks': u'console:',
      u'messaging_port': 26409,
      u'name': u'Pxps6d1C7kZ8ZuyN',
      u'tlog_compression': u'snappy',
      u'tlog_dir': u'/mnt/hdd2/arakoon/mybackend02-abm/tlogs'},
     u'global': {u'cluster': u'21Lvu16mKG7Lt3We,Pxps6d1C7kZ8ZuyN,sKa1ZvcfkiuFE0n8',
      u'cluster_id': u'mybackend02-abm',
      u'plugins': u'albamgr_plugin',
      u'tlog_max_entries': u'5000'},
     u'sKa1ZvcfkiuFE0n8': {u'client_port': 26408,
      u'crash_log_sinks': u'console:',
      u'fsync': u'true',
      u'home': u'/mnt/hdd2/arakoon/mybackend02-abm/db',
      u'ip': u'10.100.199.193',
      u'log_level': u'info',
      u'log_sinks': u'console:',
      u'messaging_port': 26409,
      u'name': u'sKa1ZvcfkiuFE0n8',
      u'tlog_compression': u'snappy',
      u'tlog_dir': u'/mnt/hdd2/arakoon/mybackend02-abm/tlogs'}},
    u'backend_guid': u'8a6fcd26-ded0-416e-a78c-03024d2f2c8b',
    u'connection_info': {u'client_id': u'1b1ac8b3-ee87-4cd8-bd1e-39bf0c24ea4b',
     u'client_secret': u'@o=>Pm|LA~!8_PBWq;#SX1pJD*xzT-P{lXF3i6Ew<yMf;,LKHJcLDWx6{z#eQ\'"{M8YLK0Js"I0dq1WN0G]RQtFeL2QEG*zv/{XEx;@k:2OM?Sj8QCDZ~rVq6}?;f4bI',
     u'host': u'10.100.199.191',
     u'local': True,
     u'port': 443},
    u'frag_size': 2097152.0,
    u'name': u'mybackend02',
    u'policies': [[1, 2, 2, 1]],
    u'preset': u'mypreset',
    u'scaling': u'LOCAL',
    u'sco_size': 4194304.0,
    u'total_size': 21420310528.0},
   u'is_backend': True,
   u'quota': False,
   u'read': True,
   u'write': False},
  u'fragment_cache': {u'backend_info': {u'alba_backend_guid': u'f6606dcf-7b8b-4842-b332-a85a33b5e4ed',
    u'arakoon_config': {u'21Lvu16mKG7Lt3We': {u'client_port': 26408,
      u'crash_log_sinks': u'console:',
      u'fsync': u'true',
      u'home': u'/mnt/hdd2/arakoon/mybackend02-abm/db',
      u'ip': u'10.100.199.191',
      u'log_level': u'info',
      u'log_sinks': u'console:',
      u'messaging_port': 26409,
      u'name': u'21Lvu16mKG7Lt3We',
      u'tlog_compression': u'snappy',
      u'tlog_dir': u'/mnt/hdd2/arakoon/mybackend02-abm/tlogs'},
     u'Pxps6d1C7kZ8ZuyN': {u'client_port': 26408,
      u'crash_log_sinks': u'console:',
      u'fsync': u'true',
      u'home': u'/mnt/hdd2/arakoon/mybackend02-abm/db',
      u'ip': u'10.100.199.192',
      u'log_level': u'info',
      u'log_sinks': u'console:',
      u'messaging_port': 26409,
      u'name': u'Pxps6d1C7kZ8ZuyN',
      u'tlog_compression': u'snappy',
      u'tlog_dir': u'/mnt/hdd2/arakoon/mybackend02-abm/tlogs'},
     u'global': {u'cluster': u'21Lvu16mKG7Lt3We,Pxps6d1C7kZ8ZuyN,sKa1ZvcfkiuFE0n8',
      u'cluster_id': u'mybackend02-abm',
      u'plugins': u'albamgr_plugin',
      u'tlog_max_entries': u'5000'},
     u'sKa1ZvcfkiuFE0n8': {u'client_port': 26408,
      u'crash_log_sinks': u'console:',
      u'fsync': u'true',
      u'home': u'/mnt/hdd2/arakoon/mybackend02-abm/db',
      u'ip': u'10.100.199.193',
      u'log_level': u'info',
      u'log_sinks': u'console:',
      u'messaging_port': 26409,
      u'name': u'sKa1ZvcfkiuFE0n8',
      u'tlog_compression': u'snappy',
      u'tlog_dir': u'/mnt/hdd2/arakoon/mybackend02-abm/tlogs'}},
    u'backend_guid': u'8a6fcd26-ded0-416e-a78c-03024d2f2c8b',
    u'connection_info': {u'client_id': u'1b1ac8b3-ee87-4cd8-bd1e-39bf0c24ea4b',
     u'client_secret': u'@o=>Pm|LA~!8_PBWq;#SX1pJD*xzT-P{lXF3i6Ew<yMf;,LKHJcLDWx6{z#eQ\'"{M8YLK0Js"I0dq1WN0G]RQtFeL2QEG*zv/{XEx;@k:2OM?Sj8QCDZ~rVq6}?;f4bI',
     u'host': u'10.100.199.191',
     u'local': True,
     u'port': 443},
    u'frag_size': 2097152.0,
    u'name': u'mybackend02',
    u'policies': [[1, 2, 2, 1]],
    u'preset': u'mypreset',
    u'scaling': u'LOCAL',
    u'sco_size': 4194304.0,
    u'total_size': 21420310528.0},
   u'is_backend': True,
   u'quota': False,
   u'read': False,
   u'write': True}}}

```
Andes:
```
In [4]: vp.metadata['backend']['caching_info']
Out[4]: 
{u'7c4c268a-ff1a-4c54-9eef-57c4a26f7d37': {u'block_cache_on_read': True,
  u'block_cache_on_write': False,
  u'fragment_cache_on_read': False,
  u'fragment_cache_on_write': True},
 u'ac0bef2b-1461-4da6-a545-2c88b2a0634b': {u'block_cache_on_read': True,
  u'block_cache_on_write': False,
  u'fragment_cache_on_read': False,
  u'fragment_cache_on_write': True},
 u'df0301fb-8121-4759-8c5b-1d900b834a8d': {u'block_cache_on_read': True,
  u'block_cache_on_write': False,
  u'fragment_cache_on_read': False,
  u'fragment_cache_on_write': True}}
```

**Comment by JeffreyDevloo** [comment]
Fixed by #2031 
Packaged in openvstorage-core_2.12.0-dev.1524492819.3d5c1e4-1_amd64.deb

**Comment by JeffreyDevloo** [comment]
Released in https://github.com/openvstorage/framework/releases/tag/2.12.0
Packaged in openvstorage-core_2.12.0-1_amd64.deb
